src/nwb_conversion_tools/tools/spikeinterface/spikeinterface.py

```diff
@@ -1132,35 +1133,40 @@ def add_units(
         )
         if property_name in ["max_channel", "max_electrode"] and nwbfile.electrodes is not None:
             unit_col_args.update(table=nwbfile.electrodes)
+
         units_table.add_column(**unit_col_args)
         aggregated_unit_properties[property_name] = checked_sorting.get_property(key=property_name)
-    for i, unit_id in enumerate(unit_ids):
-        if use_times:
-            spkt = checked_sorting.get_unit_spike_train(unit_id=unit_id, return_times=True)
-        else:
-            spkt = checked_sorting.get_unit_spike_train(unit_id=unit_id) / checked_sorting.get_sampling_frequency()
```

Comment:
Ah, so now that these objects are `BaseSorting`, they will always return in units of time (seconds)? And they have no notion of frame to timestamps mapping (the purpose of the `use_times` argument)?

Reply:
Or if they do have timestamps, they use those instead of scaling frames by frequency?

Reply:
Right, they have that implemented in house in `spikeinterface`:
https://github.com/SpikeInterface/spikeinterface/blob/12b7863684b705e3d0ae8165bb3009143c70530c/spikeinterface/core/basesorting.py#L62-L69